solver/src/solver/http_solver/model.rs

```diff
@@ -26,14 +26,16 @@ pub struct OrderModel {
 }
 
 #[derive(Debug, Serialize)]
-pub struct UniswapModel {
-    pub token1: String,
-    pub token2: String,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance1: u128,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance2: u128,
+pub struct PoolTokenData {
     #[serde(with = "serde_with::rust::display_fromstr")]
+    pub balance: u128,
+    pub weight: String,
+}
+
+#[derive(Debug, Serialize)]
+pub struct PoolModel {
+    pub pool_type: String,
+    pub reserves: HashMap<String, PoolTokenData>,
```

Comment:
As a meta comment, I'm also not a fan of `String` here as well as it causes ambiguity: what is the reserves mapping key? Token name? Token address?

Instead we should use concrete types (like `H160` or some `TokenId` type) to avoid type erasure.

I see that this is done in other parts of the HTTP solver model, so this is something that can also be done in a separate PR.

Reply:
I would much rather use H160 for all tokens. As long as it matches with the `tokens` field provided in the dictionary it should be fine on the solver side anyway.

Reply:
I have learned that the String we are using for tokens is actually just their index in an array (which is much less information than H160). Are we sure we want to change this?

Reply:
https://github.com/gnosis/gp-v2-services/pull/763